### PR TITLE
Draft: CIP 33: Multi Currency Support

### PR DESCRIPTION
**Pull request by sissnad**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1 @@
+.DS_Store
```
Comment (YazzyYaz) [inline .gitignore:1]: Can we delete this file?

File: CIPs/cip-0033.md [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+
+The addition of cEUR is the first step towards an ecosystem on Celo that hosts a family of stabilized value assets. Further assets can be added following the paragon of cEUR. Therefore, the addition of cEUR serves two purposes: extending the Celo network with an important stabilized value digital asset as well preparing the ground for future extensions with other stabilized value assets.
+
+## **Specification**
+
+### Deployment of cEUR
+
+**For StableToken:**
+
+- Retain stableToken (for compatibility).
```
Comment (martinvol) [inline CIPs/cip-0033.md:37]:
```suggestion
- Retain stableToken in the registry (for compatibility).
```

File: CIPs/cip-0033.md [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+- Create a new entry called cUSD pointing to the same as stableToken entry.
+- Create a new entry called cEUR pointing to a new cEUR proxy.
+- Add a new Gas currency (cEUR) to the FeeCurrencyWhitelist.
+
+**Proxy implementations:**<br>
+A priori, no changes should be written to the proxy contracts.
+
+- Deploy a new cUSD proxy (e.g. cUSDProxy) pointing to the same implementation as existing stableTokenProxy.
+- Deploy a cEUR proxy (e.g. cEURProxy).
```
Comment (martinvol) [inline CIPs/cip-0033.md:46]:
```suggestion
**Implementation**
- Deploy a cEUR proxy (e.g. cEURProxy), with the same implementation as cUSDProxy.
```

File: CIPs/cip-0033.md [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+
+### Stability Mechanism
+
+For cEUR one independent Mento will be deployed. All Mento exchanges (cUSD and cEUR) will be backed by the same reserve.
+
+**For Exchange:**
+
+- Retain exchange (for compatibility).
```
Comment (martinvol) [inline CIPs/cip-0033.md:54]:
```suggestion
- Retain exchange in the registry (for compatibility).
```

File: CIPs/cip-0033.md [hunk 5]
```diff
@@ -0,0 +1,164 @@
+---
+cip: 33
+title: Multi Currency Support
+author: Nadiem Sissouno (@sissnad)
```
Comment (martinvol) [inline CIPs/cip-0033.md:4]:
```suggestion
author: Nadiem Sissouno (@sissnad), Martín Volpe (@martinvol)
```

File: CIPs/cip-0033.md [hunk 6]
```diff
@@ -0,0 +1,164 @@
...
+
+**For Exchange:**
+
+- Retain exchange (for compatibility).
+- Create a new entry called (e.g. exchangecUSD) pointing to the same as exchange entry.
+- Create a new entry called (e.g exchangecEUR) pointing to a new exchange proxy for cEUR.
+
+**For Reserve:**
+
+- New Proxy spender address should be added to the reserve.
```
Comment (martinvol) [inline CIPs/cip-0033.md:60]:
```suggestion
- The reserve needs to be able to send CELO to multiple exchanges spenders (see #6165).
- New Proxy spender address should be added to the reserve.
```

File: CIPs/cip-0033.md [hunk 7]
```diff
@@ -0,0 +1,164 @@
...
+### Deployment of cEUR
+
+**For StableToken:**
+
+- Retain stableToken (for compatibility).
+- Create a new entry called cUSD pointing to the same as stableToken entry.
+- Create a new entry called cEUR pointing to a new cEUR proxy.
+- Add a new Gas currency (cEUR) to the FeeCurrencyWhitelist.
+
+**Proxy implementations:**<br>
```
Comment (YazzyYaz) [inline CIPs/cip-0033.md:42]: Delete the <br>

File: CIPs/cip-0033.md [hunk 8]
```diff
@@ -0,0 +1,164 @@
+---
+cip: 33
+title: Multi Currency Support
+author: Nadiem Sissouno (@sissnad)
+discussions-to: https://github.com/celo-org/celo-proposals/issues/125
+status: Draft
+type: Standards Track
+category: Ring 0
```
Comment (prestwich) [inline CIPs/cip-0033.md:8]: nit: I think this is `Ring 1`. Seems like a core feature that won't cause chain halt if we mess something up

File: CIPs/cip-0033.md (hunk 7 above)
Comment (YazzyYaz) [reply CIPs/cip-0033.md:42]: `<br>`

File: CIPs/cip-0033.md (hunk 8 above)
Comment (YazzyYaz) [reply CIPs/cip-0033.md:8]: Moving it to Ring 1 based on the mental model by @prestwich that describes the classification best:
```
classify by failure modes:
ring 0: halt the chain (liveness or safety violation), require manual intervention
ring 1: chain does bad things but keeps running
```

Given it's an upgrade to core contracts that doesn't halt the chain, it's a Ring 1. Will update the category and approve and merge the PR.

**Review by YazzyYaz — APPROVED** [review]

**Pull request merged**